sbtc/src/testing/regtest.rs

```diff
@@ -152,7 +152,13 @@ impl Recipient {
     /// Generate a new public-private key pair and address of the given
     /// kind.
     pub fn new(kind: AddressType) -> Self {
-        let keypair = secp256k1::Keypair::new_global(&mut rand::rngs::OsRng);
+        Self::new_with_rng(kind, &mut rand::rngs::OsRng)
+    }
+
+    /// Generate a new public-private key pair and address of the given
+    /// kind using the given random number generator.
+    pub fn new_with_rng<R: rand::Rng + ?Sized>(kind: AddressType, rng: &mut R) -> Self {
```

Comment:
Do you need ?Sized here?

Reply:
Apparently not. I originally added it to match the same type signature as `secp256k1::Keypair::new_global`, but I just removed it and was able to compile our tests just fine. So I'll update it.

Reply:
Updated in https://github.com/stacks-sbtc/sbtc/pull/1669/commits/efd458e097bcee6b3ca3bf3237186afc1a4547b8